lattice-control/lattice-control-interface.smithy

```diff
@@ -583,7 +618,7 @@ structure RegistryCredential {
     /// If supplied, token authentication will be used for the registry
     token: String,
     /// If supplied, username and password will be used for HTTP Basic authentication
-    username: String,
+    userName: String,
```

Comment:
should this be serialized to `user_name` now that it's camelCase?

Reply:
I don't think this was supposed to become camel cased since a previous PR made it `username`

Reply:
@connorsmith256 do you remember the "correct" one

Reply:
yeah, `username` is as the ascended ones gave it to us

Reply:
Yeah, for a while we had a mixture of `username` and `userName` in smithy files and the code that handles them. `userName` gets (de)serialized in rust structs as `user_name`, and we ran into issues sending data over the wire due to inconsistencies over whether we sent `userName` or `user_name` in JSON. With `username` there's no ambiguity*, so that's what I've been using everywhere.

 \* IMO username is a single compound word, i.e. we shouldn't use `userName`, because we wouldn't use `passWord` or `logIn` or `skateBoard` (fellow kids)

Reply:
Looks like the smithy file itself still has `userName`, unless I'm looking at an outdated commit

Reply:
If you use camelCase in smithy file it will turn into snake case when serialized. If we want to use the single compound word when serialized either add a @ serialized name trait, or remove the camelCase. For consistency, IMO, a single compound word shouldn't be camelcase.